azureml-minimal/cluster.tf

```diff
@@ -0,0 +1,93 @@
+data "azurerm_client_config" "current" {
+  depends_on = [azurerm_resource_group.rg]
+
+}
+
+# workspace application insights
+resource "azurerm_application_insights" "ai" {
+  name                = "${local.prefix}-${local.azureml.cluster_name}-ai"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  application_type    = "web"
+}
+
+# workspace keyvault
+resource "azurerm_key_vault" "secret_manager" {

```

Comment:
It'd be great to have the key vault resources in a separate file for easy viewing and copy/pasting in the future.
Same with the storage account.